InternalEngineTests testShouldPeriodicallyFlushAfterMerge failure

```
java.lang.AssertionError: 
Expected: <false>
     but: was <true>
	at __randomizedtesting.SeedInfo.seed([D9AEEA268A616C55:DFF296783E7FFD87]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.index.engine.InternalEngineTests.testShouldPeriodicallyFlushAfterMerge(InternalEngineTests.java:5127)
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.5+matrix-java-periodic/ES_RUNTIME_JAVA=corretto11,nodes=general-purpose/236/console
https://gradle-enterprise.elastic.co/s/k5bsk266lz67c

Reproduce
```
./gradlew ':server:test' --tests "org.elasticsearch.index.engine.InternalEngineTests.testShouldPeriodicallyFlushAfterMerge" \
  -Dtests.seed=D9AEEA268A616C55 \
  -Dtests.security.manager=true \
  -Dtests.locale=ky \
  -Dtests.timezone=Etc/GMT-14 \
  -Dcompiler.java=12 \
  -Druntime.java=11
```

--- a/server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java
+++ b/server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java
@@ -4754,6 +4754,7 @@ public class InternalEngineTests extends EngineTestCase {
 
     public void testShouldPeriodicallyFlushAfterMerge() throws Exception {
         engine.close();
+        // Do not use MockRandomMergePolicy as it can cause a force merge performing two merges.
         engine = createEngine(copy(engine.config(), newMergePolicy(random(), false)));
         assertThat("Empty engine does not need flushing", engine.shouldPeriodicallyFlush(), equalTo(false));
         ParsedDocument doc =

--- a/server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java
+++ b/server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java
@@ -4753,6 +4753,8 @@ public class InternalEngineTests extends EngineTestCase {
     }
 
     public void testShouldPeriodicallyFlushAfterMerge() throws Exception {
+        engine.close();
+        engine = createEngine(copy(engine.config(), newMergePolicy(random(), false)));
         assertThat("Empty engine does not need flushing", engine.shouldPeriodicallyFlush(), equalTo(false));
         ParsedDocument doc =
             testParsedDocument(Integer.toString(0), null, testDocumentWithTextField(), SOURCE, null);

